packages/editor/src/utils/media-upload/index.js

```diff
@@ -32,20 +37,38 @@ export default function mediaUpload( {
 	onFileChange,
 } ) {
 	const { getCurrentPostId, getEditorSettings } = select( editorStore );
+	const { lockPostSaving, unlockPostSaving } = dispatch( editorStore );
```

Comment:
I don't believe that these disable the publish button, you may need to add `enablePublishSidebar` and `disablePublishSidebar` as well. This may be overkill as the upload might not take that long but worth looking at.

Reply:
they should! based on https://github.com/WordPress/gutenberg/blob/trunk/packages/editor/src/components/post-publish-button/index.js#L121. I can double check that this works as expected and capture a screencast.

Reply:
Just tested this and the publish button is not deactivated `disablePublishSidebar` is explicitly called. This is very odd because if I call `wp.data.dispatch('core/editor').lockPostSaving()` in the console directly, the Publish button is correctly disabled.

Reply:
https://user-images.githubusercontent.com/1259027/189547590-067a4d54-7a69-4bf3-9fe9-19900ff3c0d9.mp4

Screen capture attached.

Reply:
Maybe caused by a different part of the conditional there? I'll try to debug further.